plugins/plugin-flexsearch/src/client/composables/useSearchIndex.ts

```diff
@@ -0,0 +1,30 @@
+import { searchIndex as searchIndexRaw } from '@internal/searchIndex'
+import FS from 'flexsearch'
+import { ref } from 'vue'
+
+export interface SearchIndexRet {
+  path: [string, string]
+  title: string
+}
+
+export type CSearchIndex = (string, number) => SearchIndexRet[]
+
+const index = FS.create({
+  async: false,
+  doc: {
+    id: 'id',
+    field: ['title', 'content'],
+  },
+})
+index.import(searchIndexRaw.idx)
+
+export const useSearchIndex = ref((q: string, c: number) => {
```

Comment:
I am confused why a function ref is created here

Reply:
tbh because I don't know vue all that much, and this is pretty much a copy, paste, adapt based development :)